src/software/ai/navigator/path_planner/enlsvg_path_planner.h

```diff
@@ -0,0 +1,152 @@
+#include <queue>
+
+#include "extlibs/enlsvg/Pathfinding/ENLSVG.h"
+#include "path_planner.h"
+#include "software/world/world.h"
+
+/**
+ * The Edge N-Level Sparse Visibility Graph algorithm is a fast pathfinding algorithm for
+ * precomputed grids. This class is a wrapper for an implementation found in
+ * software/src/extlibs/enlsvg/. This slide deck from Ohohcaketester
+ * [https://github.com/Ohohcakester/Any-Angle-Pathfinding/files/1286724/ENLSVG_SoCS17.pdf]
+ * and this research paper [https://aaai.org/ocs/index.php/SOCS/SOCS17/paper/view/15790]
+ * provides more information.
+ *
+ * Since this algorithm requires precomputation to achieve fast pathfinding, obstacles are
+ * passed into the constructor and is unmodifiable once it is created.
+ */
+class EnlsvgPathPlanner : public PathPlanner
+{
+   public:
+    /**
+     * Creates an EnlsvgPathPlanner object. It creates an internal grid representation
+     * using the passed navigable_area with the passed resolution. It also pads the outer
+     * boundaries of the grid as an obstacle using the boundary_margin_offset parameter.
+     * It constructs the grid and sets the corresponding obstacles as "blocked".
+     *
+     * @param navigable_area         The total area that the path planner must path plan
+     * inside
+     * @param obstacles              A list of obstacles to consider when path planning
+     * @param boundary_margin_offset Ppadding from the edges of the navigable_area to set
+     * as an obstacle
+     * @param resolution             The resolution of the internal representation of the
+     * grid. Uses the same units as boundary_margin_offset
+     */
+    EnlsvgPathPlanner(const Rectangle &navigable_area,
+                      const std::vector<ObstaclePtr> &obstacles,
+                      double boundary_margin_offset = 0.0, double resolution = 0.09);
+    /**
+     * Returns a path that is an optimized path between start and end.
+     *
+     * @param start start point
+     * @param end end point
+     * @param navigable_area Rectangle representing the navigable area
+     * @param obstacles ignored
+     *
+     * @return a vector of points that is the optimal path avoiding obstacles
+     *         if no valid path then return empty vector
+     */
+    std::optional<Path> findPath(const Point &start, const Point &end,
+                                 const Rectangle &navigable_area,
+                                 const std::vector<ObstaclePtr> &obstacles) override;
```

Comment:
you can actually not give it a name
```suggestion
                                 const std::vector<ObstaclePtr> &) override;
```

Reply:
fixed